docs/DeveloperGuide.md

```diff
@@ -1,38 +1,312 @@
+    * [Product scope](#product-scope)
+    * [Target user profile](#target-user-profile)
+    * [Value proposition](#value-proposition)
+    * [User Stories](#user-stories)
+    * [Non-Functional Requirements](#non-functional-requirements)
+    * [Glossary](#glossary)
+
+---
+
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries --
+include links to the original source as well}
+
+---
+
+## Setting up, getting started
+
+---
+
+## Design
+
+* [Architecture](#architecture)
+* [UI Component](#ui-component)
+* [Parser Component](#parser-component)
+* [Recipe Component](#recipe-component)
+* [RecipeList Component](#recipelist-component)
+* [WeeklyPlan Component](#weeklyplan-component)
+* [Database Component](#database-component)
+
+### Architecture
+
+
+### UI Component
+API: `Ui.java`
+
+### Parser Component
+API: `Parser.java`
+
+### Recipe Component
+API: `Recipe.java`
+
+The `Recipe` component:
+
+* contains `name` and `ingredients` attribute
+* store the ingredients details the user has added in `ingredients`
+
+### RecipeList Component
+
+API: `RecipeList.java`
+
+The `RecipeList` component:
+
+* extends from `ArrayList<Recipe>`
+* stores the recipes the user has added as `Recipe` objects
+* allows users to add their own recipes
+* allows users to delete the existing recipes
+* allows users to edit the existing recipes
+* allows users to view the entire list of recipes
+* allows users to view the ingredients required for a recipe
+* allows users to add and remove recipes from a tag
+* allows users to random a recipe from all the recipes that users have
+
+### WeeklyPlan Component
+
+API: `WeeklyPlan.java`
+
+The `WeeklyPlan` component:
+
+* extends from `HashMap<String, Integer>`
+* stores the names of recipes that the user plans to prepare for the week as `String`
+* stores the number of times the user plans to prepare each recipe as `Integer`
+* allows users to add single or multiple recipes from the weekly plan
+* allows users to delete single, multiple, or all recipes from the weekly plan
+
+How the `WeeklyPlan` component works:
+
+1. When the user enters an input with the first word being `weekly`, the input is passed to
+   the `Parser` component.
+2. Based on the second argument, the `Parser` component will call the appropriate method in the
+   `Parser` component.
+3. `Parser` component then returns a `WeeklyPlan` object to indicate the recipe(s) to be added or
+   deleted from the weekly plan.
+4. `WeeklyPlan` component then uses either `addPlans` or `deletePlans` method to add or delete
+   the recipe(s) from the weekly plan.
+
+The sequence diagram below shows how the `WeeklyPlan` component works when the user
+inputs `weekly /add burger 1`:
+![](../docs/UML/WeeklyPlan/AddWeeklyPlanUML.png)
```

Comment:
The diagram cannot be seen on the developer website. It might be good to review and see if this is the intended effects.